web/src/ShareSnapshotModal.tsx

```diff
@@ -0,0 +1,59 @@
+import React from "react"
+import { Snapshot } from "./types"
+import "./ShareSnapshotModal.scss"
+
+type props = {
+  handleSendSnapshot: (snapshot: Snapshot) => void
+  handleClose: () => void
+  show: boolean
+  state: Snapshot
+  snapshotUrl: string
+  registerTokenUrl: string
+}
+export default function ShareSnapshotModal(props: props) {
+  const showHideClassName = props.show
+    ? "modal display-block"
+    : "modal display-none"
+  const hasLink = props.snapshotUrl !== ""
+  if (!hasLink) {
+    return (
+      <div className={showHideClassName}>
+        <section className="modal-main">
```

Comment:
this isn't what `section` is for, see 
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/section#Usage_notes

```
Do not use the <section> element as a generic container; this is what <div> is for, especially when the sectioning is only for styling purposes. A rule of thumb is that a section should logically appear in the outline of a document.
```